src/sentry/search/events/builder.py

```diff
@@ -111,14 +115,18 @@ def __init__(
         ) = self.load_config()
 
         self.limitby = self.resolve_limitby(limitby)
-        self.where, self.having = self.resolve_conditions(
-            query, use_aggregate_conditions=use_aggregate_conditions
-        )
-        # params depends on parse_query, and conditions being resolved first since there may be projects in conditions
-        self.where += self.resolve_params()
-        self.columns = self.resolve_select(selected_columns, equations)
-        self.orderby = self.resolve_orderby(orderby)
-        self.array_join = None if array_join is None else [self.resolve_column(array_join)]
+
+        # TODO(wmak): Refactor snuba/metrics.py to call QueryBuilder correctly.
+        if not partial:
```

Comment:
Instead of a `partial` param, what if the constructor doesn't do any of the resolutions automatically? Instead, we can move this into a helper like

```python
def resolve_everything(self):
  # do the resolution here can
  # this can be overridden by child classes
  # (i think trends needs this) to change up how the resolution works
```
Then the `run_query` proposed in #31354 can just call it.

This way, you don't need to special case the uses where they just want to use the builder to resolve a few things and we still get the same convenient interface to work with. Thoughts?

Reply:
Yeah I had the same idea and talked about it with @wmak and he brought up a good point that it's kinda nice to be able resolve it in the constructor because it gives the caller the flexibility to add to or remove from these attributes (looks like the trends functions do that because we resolve the trends columns outside the query builder [here](https://github.com/getsentry/sentry/blob/ac6daa30918636b2a38bea735e7158e96f876e1c/src/sentry/api/endpoints/organization_events_trends.py#L477-L489))

Reply:
But yeah this partial thing kinda feels hacky

Reply:
I'm not sure I agree with it being nice that you are able to add/remove these attributes outside of the query builder. I'm also guilty of doing this in the suspect spans query where I append additional conditions. While convenient, it moves some of the responsibility of building the query outside of the query builder. It also leads to hacks like that in the trends query builder where it doesn't specify an order by in the constructor only to manually add one later on.

In this world where we have this `resolve_everything` call, we are able to subclass the query builder and implement a custom query resolution that can include adding/removing from the query attributes.

Reply:
I like the idea of another class with an empty `resolve_everything`. If the class is named appropriately, this would cater to both use cases in a clean way.

Reply:
Oooh yeah maybe let me try it out with a "resolve_query" method

Reply:
oh yeah `resolve_query` is a far better method name :+1: